eth/downloader/statesync.go

```diff
@@ -208,6 +210,37 @@ func (d *Downloader) runStateSync(s *stateSync) *stateSync {
 			active[req.peer.id] = req
 		}
 	}
+	// We're exiting now, but we need to time out (or handle outstanding requests first)
+	for len(active) > 0 {
```

Comment:
Would be a little cleaner to put this loop into a function 

```
func (d *Downloader) spindownStateSync(active map[string]*stateReq, s, next *stateSync) *stateSync
```

This would allow you to just `return d.spindownStateSync(active, s, next)` above, instead of tracking
the return value across two loops in a variable.

Reply:
Neat!

Reply:
Well, actually, it's not quite as nice as that, since I'll also need to pass it hte `peerDrop` and `timeout` chans

Reply:
Fixed